core/partitioning/partitioning.cpp

```diff
@@ -145,254 +94,26 @@ void getDirtyNodes(
         }
-        pytorch_nodes.push_back(n);
-      } else {
-        if (!pytorch_nodes.empty()) {
-          new_seg_blocks.emplace_back(new_seg_blocks.size(), SegmentedBlock::kTorch, pytorch_nodes);
-          pytorch_nodes.clear();
-        }
-        tensorrt_nodes.push_back(n);
-      }
-    }
-
-    // Form the last segmented_block with the leftover nodes in tensorrt_nodes or pytorch_nodes correspondingly.
-    if (!tensorrt_nodes.empty()) {
-      new_seg_blocks.emplace_back(new_seg_blocks.size(), SegmentedBlock::kTensorRT, tensorrt_nodes);
-    } else {
-      new_seg_blocks.emplace_back(new_seg_blocks.size(), SegmentedBlock::kTorch, pytorch_nodes);
-    }
-  }
-
-  return new_seg_blocks;
-}
-
-PartitionedGraph segmentBlocksWithNonTensorInputs(SegmentedBlock& seg_block) {
-  // reconstruct segmented_block if this block requires nonTensor input
-  std::vector<torch::jit::Value*> inputs_to_resolve;
-  // Gather all non-tensor inputs for this block
-  for (auto input : seg_block.raw_inputs()) {
-    if (!isTensorOrTensorList(input)) {
-      inputs_to_resolve.push_back(input);
-    }
-  }
-  return segmentBlocksWithSpecifiedInputs(seg_block, inputs_to_resolve);
-}
-
-std::unordered_map<torch::jit::Value*, usage_info> getInputUsageCounts(
-    const PartitionedGraph& segmented_blocks,
-    const std::function<bool(torch::jit::Value*)>& condition) {
-  // usage_counts is a map which stores non-tensor inputs as keys and the values are indices of segmented blocks which
-  // have these non-tensor inputs. Iterate through the graph (segmented blocks) from bottom to top. When we find a
-  // non-tensor input in a segmented block of index "i", store it in the usage_counts map. Now for each non-tensor
-  // inputs recorded in the usage_counts map, we check if any previous segmented block (segmented block index i goes
-  // from n-1 to 0) generated/contains this non-tensor input. If so, we set this idx as the produce_id as it produces
-  // the non-tensor input.
-  std::unordered_map<torch::jit::Value*, usage_info> usage_counts;
-  for (int i = segmented_blocks.size() - 1; i >= 0; --i) {
-    for (auto input : segmented_blocks[i].raw_inputs()) {
-      if (condition(input)) {
-        segmented_blocks[i].target() == SegmentedBlock::kTorch ? usage_counts[input].torch_use_id.push_back(i)
-                                                               : usage_counts[input].tensorrt_use_id.push_back(i);
       }
-    }
-
-    // For each non-tensor value in the usage_counts map, keep updating the produce_id to the earliest segmented block
-    // that has/produces it.
-    for (auto& use : usage_counts) {
-      // Set the produce_id to the segmented block index that contains/produces this non-tensor torch::jit::Value
-      if (segmented_blocks[i].contain_raw_value(use.first)) {
-        use.second.produce_id = i;
+      if (!inputs_to_resolve.empty()) {
+        std::vector<torch::jit::Node*> dependency_nodes = getDependencyNodes(inputs_to_resolve);
```

Comment:
getDependencyNodes will miss any non-modifying dependency ops as it is currently written.

Reply:
thanks @mfeliz-cruise ! 
Let me check if there is any APIs that we can use in Torchscript to find all the modifying ops as well.